muikku/src/main/webapp/resources/scripts/gui/guider.js

```diff
@@ -21,34 +21,65 @@
     
     _onFilterLink: function (event) {
       var element = $(event.target).closest('.gu-filter-link');
-      $('.gt-students-view-container').guiderStudents('workspaces', [ element.attr('data-id') ]);
+      switch (element.attr('data-type')) {
```

Comment:
Why not "if" and ".guiderStudents(element.attr('data-type'), [...])" or something like it?